### PR TITLE
Ameerul / FEQ-2453 Show a message if the user is banned from using P2P

### PR DESCRIPTION
**Pull request by ameerul-deriv**
- Added flows for Permission Denied, perm banned user, cashier is locked, system maintenance
- Added new custom hook useIsP2PBlocked
- Updated content in BlockedScenarios

Screenshot 2024-07-10 at 2 39 49 PM
Screenshot 2024-07-10 at 2 50 55 PM

https://github.com/deriv-com/p2p/assets/103412909/0a19542e-662d-4879-936a-64852e05f68b

https://github.com/deriv-com/p2p/assets/103412909/3c3adee0-046c-46ec-82d9-31ac99986372

https://github.com/deriv-com/p2p/assets/103412909/1dafa2a5-b13d-4fbe-b370-0690eba2c9f8

https://github.com/deriv-com/p2p/assets/103412909/1d17ca07-bca0-49d8-858a-829a075d2135

https://github.com/deriv-com/p2p/assets/103412909/259defa7-d2c2-4c25-a479-2967a0b207a7

### CODE REVIEW

File: src/components/BlockedScenarios/BlockedScenarios.tsx [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
     // TODO: change redirection when account switcher is implemented
     const openDerivApp = () => {
         window.open(URLConstants.derivAppProduction, '_blank')?.focus();
     };
 
+    // TODO: implement this function to open live chat
+    const openLiveChat = () => {};
+
     const blockedScenarios: TBlockedScenariosObject = {
+        cashierLocked: {
```
Comment (ameerul-deriv) [inline src/components/BlockedScenarios/BlockedScenarios.tsx:35]: i asked yunus what we could show here, and he had shown me this:

Screenshot 2024-07-10 at 2 56 37 PM

File: src/components/BlockedScenarios/BlockedScenarios.tsx [hunk 2]
```diff
@@ -77,16 +104,60 @@ const BlockedScenarios = ({ type }: { type: string }) => {
...
                     icon={blockedScenarios[type].icon}
                     title={blockedScenarios[type].title}
                 />
+            ) : (
+                <ActionScreen
+                    actionButtons={blockedScenarios.p2pBlocked.actionButton}
+                    description={blockedScenarios.p2pBlocked.description}
+                    icon={blockedScenarios.p2pBlocked.icon}
+                    title={blockedScenarios.p2pBlocked.title}
+                />
```
Comment (ameerul-deriv) [inline src/components/BlockedScenarios/BlockedScenarios.tsx:160]: by default in v1, we show this if permission denied happens, so I have this return as the default blocked screen when user gets permission denied

File: src/hooks/custom-hooks/useIsP2PBlocked.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { api } from '..';
+
+// TODO: add other blocked statuses when implementing high risk scenarios
+/**
+ * A custom hook that returns the P2P blocked status and the status type.
+ * @returns {object} { isP2PBlocked: boolean, status: string }
+ */
+const useIsP2PBlocked = () => {
+    const { data: accountStatus } = useGetAccountStatus();
+    const { data: activeAccountData } = api.account.useActiveAccount();
```
Comment (ameerul-deriv) [inline src/hooks/custom-hooks/useIsP2PBlocked.ts:13]: i also checked the network tab, this does not call active account list multiple times

File: src/components/BlockedScenarios/BlockedScenarios.tsx (hunk 2 above)
Comment (nada-deriv) [reply src/components/BlockedScenarios/BlockedScenarios.tsx:160]: why not give type='p2pBlocked' as default?

Comment (ameerul-deriv) [reply src/components/BlockedScenarios/BlockedScenarios.tsx:160]: let me try this 👀

File: src/hooks/custom-hooks/useIsP2PBlocked.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        if (accountStatus.status.includes('cashier_locked')) return 'cashierLocked';
+        if (accountStatus.p2p_status === 'perm_ban') return 'p2pBlocked';
+
+        return '';
+    }, [accountStatus]);
+
+    const isP2PNotSupported = useMemo(() => {
+        if (!activeAccountData) return false;
+
+        return activeAccountData.is_virtual || activeAccountData.currency !== 'USD';
```
Comment (nada-deriv) [inline src/hooks/custom-hooks/useIsP2PBlocked.ts:38]: isn't this check already handled in getBlockedType?

File: src/hooks/custom-hooks/useIsP2PBlocked.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    const blockedType = useMemo(
+        () => (activeAccountData ? getBlockedType(activeAccountData) : ''),
+        [activeAccountData]
+    );
+
+    const isP2PBlocked = Boolean(isAccountBlocked || isP2PNotSupported);
+    const status = isAccountBlocked ? accountBlockStatus : blockedType;
+
+    return { isP2PBlocked, status: status || '' };
```
[the remark this answers is not in the document]
Comment (ameerul-deriv) [reply src/hooks/custom-hooks/useIsP2PBlocked.ts:49]: true, i can check this

File: src/hooks/custom-hooks/useIsP2PBlocked.ts (hunk 4 above)
Comment (ameerul-deriv) [reply src/hooks/custom-hooks/useIsP2PBlocked.ts:38]: for this, we are using this to determine whether or not we show BlockedScenarios component using isP2PBlocked flag. though perhaps we could just check if type was returned from blockedType.

Comment (nada-deriv) [reply src/hooks/custom-hooks/useIsP2PBlocked.ts:38]: yeah.. is_virtual will have demo and the other will have nonUSD. so u can just check for the type.. so i dont think this function is needed.. can check for the blockedType value where this function is used

File: src/hooks/custom-hooks/useIsP2PBlocked.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ameerul-deriv) [reply src/hooks/custom-hooks/useIsP2PBlocked.ts:49]: ok for here i wont return p2pBlocked as the default, ill leave it as an empty string, and let the component have the default value instead

**Review by nada-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**